R/module_nested_tabs.R

```diff
@@ -247,24 +254,56 @@ srv_teal_module.teal_module <- function(id,
     filtered_teal_data <- eventReactive(trigger_data(), {
       .make_teal_data(modules, data = data_rv(), datasets = datasets(), datanames = active_datanames())
     })
-    srv_data_summary("data_summary", filtered_teal_data)
+
+    transformed_teal_data <- srv_teal_data_modules(
+      "data_transform",
+      data = filtered_teal_data,
+      transformers = modules$transformers,
+      modules = modules
+    )
```

Comment:
This needs to be wrapped on a `eventReactive` just as chunk above. Otherwise, non-DDL applications fail on init

Simple app that reproduces this

```r

options(
  teal.log_level = "ERROR",
  teal.show_js_log = TRUE,
  # teal.bs_theme = bslib::bs_theme(version = 5),
  shiny.bookmarkStore = "server"
)

pkgload::load_all("../teal")
utils::data(miniACC, package = "MultiAssayExperiment")

data <- teal_data() |> 
  within({
    utils::data(miniACC, package = "MultiAssayExperiment", envir = environment())
    MAE <- miniACC
  })

datanames(data) <- "MAE"


data <- teal_data() |>
  within(IRIS <- iris)

datanames(data) <- "IRIS"
init(
  data = data,
  modules = list(
    example_module(
      transformers = list(
        teal_data_module(
          label = "head",
          ui = function(id) {
            tags$span("")
          },
          server = function(id, data) {
            checkmate::assert_class(data, "reactive")
            checkmate::assert_class(isolate(data()), "teal_data")
            moduleServer(id, function(input, output, session) {

              data
            })
          }
        )
      )
    )
  )
) |>
  shiny::runApp()
```

```suggestion
    transformed_teal_data <- eventReactive(filtered_teal_data(), {
       new_transformed_teal_data <- srv_teal_data_modules(
        "data_transform",
        data = filtered_teal_data,
        transformers = modules$transformers,
        modules = modules
      )
      new_transformed_teal_data()
    })
```